source/lib.ts

```diff
@@ -119,39 +128,37 @@ const parseOptions = (
 			_next: NextFunction,
 			_optionsUsed: Options,
 		): void => {
-			response.status(options.statusCode).send(options.message)
+			response.status(config.statusCode).send(config.message)
 		},
 		onLimitReached: (
 			_request: Request,
 			_response: Response,
 			_optionsUsed: Options,
 		): void => {},
-		// Allow the above to be overriden by the options passed to the middleware
-		...passedOptions,
+		// Allow the options object to be overriden by the options passed to the middleware.
+		...notUndefinedOptions,
+		// Note that this field is declared after the user's options are spread in,
+		// so that this field doesn't get overriden with an un-promisified store!
+		store: promisifyStore(notUndefinedOptions.store ?? new MemoryStore()),
 	}
 
 	// Ensure that the store passed implements the either the `Store` or `LegacyStore`
 	// interface
```

Comment:
This comment is now outdated

```suggestion
	// Ensure that the store passed implements the either the `Store` interface
```

(unless we want to move this check to earlier, so that it can validate a legacy store before it's promisified)

Reply:
Resolved